src/dwd_dynam.jl

```diff
@@ -273,11 +311,14 @@ readout(f::AbstractMachine, u::FinDomFunction, args...) = readout(f, collect(u),
 
 readout(m::Machine{T,I,S}, u::AbstractVector, x::AbstractVector, p=nothing, t=0) where {T, I<:InstantaneousDirectedInterface{T}, S} = 
```

Comment:
You could probably combine these definitions of `readout` by specifying `I <: AbstractInstantaneousDirectedInterface{T}`